botocore/handlers.py

```diff
@@ -519,6 +519,24 @@ def change_get_to_post(request, **kwargs):
         request.url, request.data = request.url.split('?', 1)
 
 
+def set_list_objects_encoding_type_url(params, **kwargs):
+    if 'EncodingType' not in params:
+        # We set this context so that we know it wasn't the customer that
+        # requested the encoding.
+        kwargs['context']['EncodingTypeAutoSet'] = True
```

Comment:
You can also just accept `context` as a function param so you don't have to pull it from `**kwargs`.  `**kwargs` was really just for kwargs you don't care about and to future proof event handlers.